specs/order_spec.rb

```diff
@@ -73,38 +77,95 @@
       order = Grocery::Order.new(1337, products)
 
       result = order.add_product("salad", 4.25)
+
       result.must_equal true
     end
   end
+
+  describe "#remove_product" do
+    it "Removes the product from the list" do
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      order = Grocery::Order.new(1337, products)
+
+      order.remove_product("banana")
+
+      count = 0
+      products.each do |single_order|
+        count += 1
+      end
+      count.must_equal 1
+    end
+  end
+
+  it "Returns true if the product was present and removed" do
+    products = { "banana" => 1.99, "cracker" => 3.00 }
+    order = Grocery::Order.new(1330, products)
+
+    value = order.remove_product("banana")
+
+    value.must_equal true
+  end
+
+  it "Returns false if the product was absent and not removed" do
+    products = { "banana" => 1.99, "cracker" => 3.00 }
+    order = Grocery::Order.new(1330, products)
+
+    value = order.remove_product("pear")
+
+    value.must_equal false
+  end
 end
 
-# TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Order Wave 2" do
+describe "Order Wave 2" do
   describe "Order.all" do
     it "Returns an array of all orders" do
-      # TODO: Your test code here!
+      orders_entered = Grocery::Order.all
+
+      orders_entered.class.must_equal Array
+      count = 0
+      orders_entered.each do |order|
```

Comment:
You don't need a loop here for counting, you can just do:  

```ruby
Grocery::Order.all.length.must_equal 100
```

Just use length!